### PR TITLE
Refactor diddoc, extract as crate, create crate vcx_shared

### PR DESCRIPTION
**Pull request by Patrik-Stas**
Note: The commits in these PR are quite well separated, might we worthy to review commit-by-commit approach

This PR is further addressing https://github.com/hyperledger/aries-vcx/issues/697

Changes:
- Aries/w3c diddoc divergence
   - Added `W3cDidDoc` - this reflects https://www.w3.org/TR/did-core/#example-30-did-document-with-1-verification-method-type but is not currently used anywhere as there's a slight divergence due aries `service` extensions.
   - Renamed original `DidDoc` to `AriesDidDoc` 
   - The difference between the 2 is most of all `service` attribute - the rendered diddoc resolved by `did:sov` has an extra attributes like `recipientKeys`, `routingKeys` etc. (see `Resolver DID Document Format` @ https://sovrin-foundation.github.io/sovrin/spec/did-method-spec-template.html)
   - Long term we should use `W3cDidDoc` or find better abstraction to fit in the aries stuff
   
- Renamed `ServiceResolvable` - which sounds like and interface but is enum, to `ServiceOob`. This represents either `AriesService` or a DiD, for which AriesService can be resolved from ledger per OOB spec https://github.com/hyperledger/aries-rfcs/blob/main/features/0434-outofband/README.md#messages
   > services - an array of union types that the receiver uses when responding to the message. Each item is either a DIDComm service object (as per [RFC0067](https://github.com/hyperledger/aries-rfcs/blob/main/features/0067-didcomm-diddoc-conventions/README.md#service-conventions)) or a DID (as per [Decentralized Identifiers v1.0](https://w3c.github.io/did-core/#did-syntax)). Additional details below.
   
- Moved `EndpointDidSov` under `aries-vcx/common/ledger` - it has nothing to do with Aries messages, eventually should end up in some DID DOC resolver module
- Added optional field `types` which was missing in `EndpointDidSov` (as specified in `DID Service Endpoint` @ https://sovrin-foundation.github.io/sovrin/spec/did-method-spec-template.html#crud-operation-definitions )
- Extracted all diddoc related code to separate crate - as by itself is more general concept and has no relation to `aries`
- As result, I also ended up creating `vcx_shared` crate to avoid yet another duplication of methods `validate_did` and `validate_verkey`

Built on top of https://github.com/hyperledger/aries-vcx/pull/702

### CODE REVIEW

File: aries_vcx/src/common/ledger/service_didsov.rs [hunk 1]
```diff
@@ -1,14 +1,30 @@
...
-// Note that is divergence from w3c spec https://w3c.github.io/did-core/#service-properties
-#[derive(Debug, Deserialize, Serialize, Clone,PartialEq)]
+// https://sovrin-foundation.github.io/sovrin/spec/did-method-spec-template.html
+#[derive(Debug, Deserialize, Serialize, Clone)]
+#[cfg_attr(test, derive(PartialEq))]
 pub struct EndpointDidSov {
     pub endpoint: String,
     #[serde(default)]
-    #[serde(rename = "routingKeys")]
+    #[serde(rename = "camelCase")]
```
Comment (gmulhearn) [inline aries_vcx/src/common/ledger/service_didsov.rs:11]: will this behave correctly? i.e. will it be literally `"camelCase"`

**Review by gmulhearn — CHANGES_REQUESTED** [review]
looks good, just the one comment i wasn't sure about

File: aries_vcx/src/common/ledger/service_didsov.rs (hunk 1 above)
Comment (Patrik-Stas) [reply aries_vcx/src/common/ledger/service_didsov.rs]: uhh this was great catch, signifying importance of de/serialization tests - these has now been added

**Review by gmulhearn — DISMISSED** [review]

**Comment by Patrik-Stas** [comment]
gmulhearn thanks for the review! 

Rebased and merging

**Pull request merged**